lambdas/data-migration1/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import AWS from 'aws-sdk';
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { connection } from '@cumulus/db';
+import Logger from '@cumulus/logger';
+
+const {
+  Manager,
+} = require('@cumulus/api/models');
+const schemas = require('@cumulus/api/models/schemas');
+
+const logger = new Logger({ sender: '@cumulus/data-migration' });
+
+export interface HandlerEvent {
+  env?: NodeJS.ProcessEnv
+}
+
+export interface RDSCollectionRecord {
+  name: string
+  version: string
+  process: string
+  granuleIdValidationRegex: string
+  granuleIdExtractionRegex: string
+  files: string
+  // default will be set by schema validation
+  duplicateHandling: string
+  // default will be set by schema validation
+  reportToEms: boolean
+  sampleFileName?: string
+  url_path?: string
+  ignoreFilesConfigForDiscovery?: boolean
+  meta?: object
+  tags?: string
+  created_at: Date
+  updated_at: Date
+}
+
+const getRequiredEnvVar = (name: string, env: NodeJS.ProcessEnv): string => {
+  const value = env?.[name];
+
+  if (value) return value;
+
+  throw new Error(`The ${name} environment variable must be set`);
+};
+
+/**
+ * Migrate collection record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {number|false}
+ *   New record ID on success, false if record was skipped
+ */
+export const migrateCollectionRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<number | false> => {
+  // Use schema to validate record before processing
```

Comment:
d27c3a6bdb682102f2f48163047c802fb15868d2

Reply:
It's probably^H^H^H^H^H^H^H definitely a nit, and perhaps a bigger concern for files/granules, but the concern here was more about how we don't actually trap the error set and dump it to CW logs, so users will have to text munge that rather than having a concise list of what failed.    Success/Total might be good as well.